scrape/scrape.go

```diff
@@ -222,6 +222,7 @@ func (sp *scrapePool) stop() {
 		delete(sp.activeTargets, fp)
 	}
 	wg.Wait()
+	sp.client.Transport.(*http.Transport).CloseIdleConnections()
```

Comment:
Digging through the types feels wrong here. Could this be more of a direct method?

Reply:
It could be more direct, but I would need to propagate the method back up from `NewRoundTripperFromConfig` and then `NewClientFromConfig`. It's not a huge change, it would just require changing the signature of those methods which I was trying to avoid.

Reply:
There's a few ways that it could be done, but it should be done. This code isn't safe for example if any of the auth methods are used, as they wrap the transport.

Reply:
I could wrap it in a type check for safety. I don't really want to modify the `NewRoundTripperFromConfig` and `NewClientFromConfig` methods if I can avoid it because it would require changes in prometheus/common. What other ways are you thinking of?

Reply:
Ok, I made a PR https://github.com/prometheus/common/pull/169, what's the proper way to update go vendor? `govendor fetch ...`?

Reply:
We're using Go modules now. See https://github.com/prometheus/prometheus/blob/master/CONTRIBUTING.md#dependency-management